pkg/chart/chart.go

```diff
@@ -78,31 +81,37 @@ type HTTPClient interface {
 	Do(req *http.Request) (*http.Response, error)
 }
 
-// LoadChart should return a Chart struct from an IOReader
-type LoadChart func(in io.Reader) (*chart.Chart, error)
+type ChartMultiVersion struct {
+	V2 *chartv2.Chart
+	V3 *chartv3.Chart
+}
+
+// LoadChartV2 should return a V2 Chart struct from an IOReader
+type LoadChartV2 func(in io.Reader) (*chartv2.Chart, error)
+
+// LoadChartV3 returns a V3 Chart struct from an IOReader
+type LoadChartV3 func(in io.Reader) (*chartv3.Chart, error)
 
 // Resolver for exposed funcs
 type Resolver interface {
 	ParseDetails(data []byte) (*Details, error)
-	GetChart(details *Details, netClient HTTPClient) (*chart.Chart, error)
+	GetChart(details *Details, netClient HTTPClient) (*ChartMultiVersion, error)
 	InitNetClient(details *Details) (HTTPClient, error)
 }
 
-// Chart struct contains the clients required to retrieve charts info
-type Chart struct {
+// ChartClient struct contains the clients required to retrieve charts info
+type ChartClient struct {
 	kubeClient    kubernetes.Interface
 	appRepoClient appRepoClientSet.Interface
-	load          LoadChart
```

Comment:
AFAICT, the `LoadChart` function was only injected into the struct so that tests could use a fake? At least, that was the only place a different `LoadChart` was used... the loader was hard-coded in the `main.go`. I've instead updated the tests so that they don't need a fake, but instead load a real chart from `test_data` (I took the current `bitnami/nginx`).

I initially packaged the `nginx` chart using both helm2 and 3, and although you end up with a different tarball, tests pass with both, so I've just left the helm2 packaged one.

Reply:
It is possible to both use real data *and* modify the input - I've just not done so here because it wasn't necessary for my simple test. But if we want to in the future, we can extend this with something like the helm tests for the archive loader, in https://github.com/helm/helm/blob/master/pkg/chart/loader/archive_test.go where you generate the archive with the setup for your test.